File: frontend/src/utils/slug.ts
export function slugify(title: string): string {
  return title
    .toLowerCase()
    .trim()
    .replace(/[^a-z0-9\u4e00-\u9fa5]+/g, '-')
    .replace(/-+/g, '-')
    .replace(/^-|-$/g, '')
    .slice(0, 100);
}

export function isValidSlug(slug: string): boolean {
  return /^[a-z0-9-]+$/.test(slug) && slug.length <= 100;
}

File: backend/internal/util/slug.go
package util

import (
	"regexp"
	"strings"
)

var (
	slugRe      = regexp.MustCompile(`[^a-z0-9-]`)
	hyphenRe    = regexp.MustCompile(`-{2,}`)
)

// Slugify 标题转 slug（保留 ascii，中文转拼音占位+序号由调用方补充）。
func Slugify(title string) string {
	s := strings.ToLower(strings.TrimSpace(title))
	s = strings.ReplaceAll(s, " ", "-")
	s = slugRe.ReplaceAllString(s, "-")
	s = hyphenRe.ReplaceAllString(s, "-")
	return strings.Trim(s, "-")
}

// IsValidSlug 校验 slug 格式。
func IsValidSlug(slug string) bool {
	if slug == "" || len(slug) > 100 {
		return false
	}
	return slugRe.ReplaceAllString(slug, "") == slug
}
